apps/web/src/components/Spaces/Common/AvatarGrid/Avatar.tsx

```diff
@@ -0,0 +1,116 @@
+import { DotsVerticalIcon } from '@heroicons/react/outline';
+import { useEventListener, useHuddle01 } from '@huddle01/react/hooks';
+import { Image } from '@lenster/ui';
+import type { FC } from 'react';
+import React from 'react';
+import { useSpacesStore } from 'src/store/spaces';
+import { useUpdateEffect } from 'usehooks-ts';
+
+import Audio from '../Audio';
+import Dropdown from '../Dropdown';
+import CoHostData from '../Sidebar/Peers/PeerRole/CoHostData';
+import HostData from '../Sidebar/Peers/PeerRole/HostData';
+import ListenersData from '../Sidebar/Peers/PeerRole/ListenersData';
+import SpeakerData from '../Sidebar/Peers/PeerRole/SpeakerData';
+import type { IRoleEnum } from '../SpacesTypes';
+
+type Props = {
+  peerId?: string;
+  displayName: string;
+  mic?: MediaStreamTrack | null;
+  role?: IRoleEnum;
+  avatarUrl: string;
+};
+
+const Avatar: FC<Props> = ({ peerId, displayName, mic, role, avatarUrl }) => {
+  const { isMyHandRaised, setMyHandRaised, myReaction, setMyReaction } =
+    useSpacesStore();
+
+  const { me } = useHuddle01();
+
+  const RoleData = {
+    host: <HostData />,
+    coHost: peerId ? <CoHostData peerId={peerId} /> : null,
+    speaker: peerId ? <SpeakerData peerId={peerId} /> : null,
+    listener: peerId ? <ListenersData peerId={peerId} /> : null
+  } as const;
+
+  useEventListener('room:data-received', (data) => {
```

Comment:
Can we move this to enum? `room:data-received` or import from the SDK if exposed